.github/workflows/bench.yml

```diff
@@ -0,0 +1,26 @@
+name: Benchmarks
+on:
+  pull_request:
```

Comment:
> [!NOTE]
> Do we want to run these on master or are we comfortable with this being per PR?

Reply:
What will be the cost of doing this per PR? Both time-wise and $$$ ?

Reply:
I'd be pro doing this on ever PR, if something is going to bomb the performance we probably want to know before merging.

Reply:
> What will be the cost of doing this per PR? Both time-wise and $$$ ?

@Dhaiwat10  the benchmarks took around [11 minutes](https://github.com/FuelLabs/fuels-ts/actions/runs/10754180931/job/29824507491) I can't speak to the $ cost but it looks like according to the [pricing plans](https://codspeed.io/pricing) we would get unlimited usage.  @arboleya do you have any idea about costs?

I second what you say @danielbate